### ISSUE TITLE
Adding back almost equals

### ISSUE THREAD

**Issue by snowman2** [shapely/shapely#1555]
For adding back `almost_equals`, I would like to see some more opinions of other contributors (and we can consider this for a separate PR, to move forward with this one). 
I am personally not super convinced of the value of doing adding it back. While it could be nice to have a method that normalizes by default, IMO it is still confusing to have those two slightly different methods (eg that it has a different way to specify the tolerance)., and even when it normalizes by default, I still find "almost" misleading, as it still requires exactly the same coordinates (only the order is relaxed by first normalizing).

_Originally posted by @jorisvandenbossche in https://github.com/shapely/shapely/issues/1231#issuecomment-1262172690_

**Comment by snowman2** [comment]
If `almost_equals` is to be preserved, I think it would make more sense if almost equals would provide both the functionality that `equals` provides by not requiring the coordinates to be exactly the same while also providing a way to reduce the precision of the coordinates when doing the comparison similar to the current `almost_equals` and  `equals_exact`.

One potential option could be to extract the coordinates from the current geometry and using `numpy.around` (or something equivalent) and then re-constructing them.

**Comment by tfardet** [comment]
I'm not sure I understand the point of going forward with #1231 if we go back on the deprecation of ``almost_equals``...

Are there situation where ``normalize`` would not be sufficient to detect the equality of two geometries?

If not, then keeping ``equals_exact`` as it is seems fine to me (providing we clarify the docstring a bit) since it is supposed to compare coordinates.
On the other hand, ``almost_equals``, if using normalize by default, would properly compare (approximmate) topological equality and always be true if ``equals`` is true, wouldn't it?

This is why, if we restore ``almost_equals`` (which I think would be best in terms of logic, given the function names), I don't really see the point #1231.

**Comment by jorisvandenbossche** [comment]
> On the other hand, `almost_equals`, if using normalize by default, would properly compare (approximmate) topological equality and always be true if `equals` is true, wouldn't it?

No, `normalize` only standardizes the ordering of coordinates in lines and rings (and of rings in polygons, geometries in Multi-geometries), so that if you have two geometries that only differ in this order can be compared exactly on the coordinates (basically getting the coordinates as a 2D array of each geometry and asserting that those two arrays are equal (up to a certain tolerance)). 

But it does not change anything else structurally about your geometry. So there will be lots of cases where geometries are topological/spatially equal, but still not structurally equal even after normalization.

**Comment by tfardet** [comment]
OK, but that can only happen for (Multi)Polygons, right?
Like if the order of the rings or of the Polygons is different (that's pretty much the only case I can think of).

In that case, indeed, unless there's a way to rewrite ``almost_equals`` to make sure it can solve these cases, then it does not make sense to keep it.
